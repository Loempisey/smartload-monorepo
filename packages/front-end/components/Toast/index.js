// import { success } from "concurrently/src/defaults";
import { toast } from "react-toastify";

const character = {
    position: "top-right",
    autoClose: 5000,
    hideProgressBar : false,
    closeOnclick: true,
    pauseOnHover: true,
    draggable: true,
    progress: undefined,

};

function customToast(statusCode, message) {
    console.log(statusCode)
    if (statusCode == 200 || statusCode == 201){
        toast.success(message, character);
    }else if(statusCode == 400 || statusCode == 404 || statusCode == 500){
        toast.error(message, character);
    }else{
        toast.error(message, character);
    }
}

async function customPromiseToast(fetchAPI){
    const data = await toast.promise(fetchAPI, {
        pending: "Operater is progressing",
        success: "Operater is success",
        error: "Operater is error",

    });
    customTost(data.statusCode, data?.message);
}

export{
    customToast,
    customPromiseToast,
}


