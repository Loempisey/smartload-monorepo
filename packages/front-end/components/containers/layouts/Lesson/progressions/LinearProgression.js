import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import LinearProgressWithLabel from './LinearWithLabel';


const useStyles = makeStyles({
  root: {
    width: '100%',
  },
});

export default function LinearProgression({increase=0,show=false}) {
  const classes = useStyles();
  
  return (
    <div className={classes.root}>
        {
            show&&(<LinearProgressWithLabel value={increase} />)
        }
      
    </div>
  );
}
