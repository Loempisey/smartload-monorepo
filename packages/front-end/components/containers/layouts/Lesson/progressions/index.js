import LinearProgression from "./LinearProgression";
export {
    LinearProgression
}