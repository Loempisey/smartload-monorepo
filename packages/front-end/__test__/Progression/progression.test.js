import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import {LinearProgression, LinearProgressionWithLabel} from '../../components/progressions';
import {LinearProgressionWithLabel} from '../../components/progressions/LinearWithLabel'
describe('Linear Progression', ()=>{
    it("snapshot linear",()=>{
        const wrapper = shallow(<LinearProgression/>)
        console.log(wrapper)
        expect(toJson(wrapper)).toMatchSnapshot();

    });
    it("show linear Progression",() =>{
        const wrapper = shallow(<LinearProgression show={true}/>)
        expect(wrapper.find('LinearProgressionWithLabel').exists()).toBeTruthy();


    });
    it("hide linear progression",()=>{
        const wrapper = shallow(<LinearProgression show={false}/>)
        expect(wrapper.find('LinearProgressionWithLabel').exists()).toBeFalsy();
    });
})





