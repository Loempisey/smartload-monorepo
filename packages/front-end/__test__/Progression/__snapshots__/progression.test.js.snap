// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Linear Progression snapshot linear 1`] = `
<div
  className="makeStyles-root-1"
/>
`;
